### ISSUE TITLE
Cannot install anaconda 5.3 using conda

### ISSUE THREAD

**Issue by michaellass** [conda/conda#7984]
## Current Behavior

Trying to update to or to install the anaconda package in version 5.3.0 or 5.3.1 fails with the following message:

```
UnsatisfiableError: The following specifications were found to be in conflict:
  - anaconda=5.3.1
Use "conda info <package>" to see the dependencies for each package.
```

### Steps to Reproduce

```
conda create -n anaconda anaconda=5.3.1
```
The same issue shows up for anaconda=5.3.0, anaconda=5.3.0=py37_0 or similar, or when trying to update an older anaconda installation.

## Expected Behavior

anaconda should be installed in the corresponding version. Installing anaconda=5.2.0 for example works just fine.

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>

```
     active environment : base
    active env location : /scratch/bevan/anaconda3
            shell level : 1
       user config file : /home/bevan/.condarc
 populated config files : 
          conda version : 4.5.11
    conda-build version : not installed
         python version : 3.7.1.final.0
       base environment : /scratch/bevan/anaconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/linux-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/linux-64
                          https://repo.anaconda.com/pkgs/pro/noarch
          package cache : /scratch/bevan/anaconda3/pkgs
                          /home/bevan/.conda/pkgs
       envs directories : /scratch/bevan/anaconda3/envs
                          /home/bevan/.conda/envs
               platform : linux-64
             user-agent : conda/4.5.11 requests/2.20.1 CPython/3.7.1 Linux/4.19.2-arch1-1-ARCH arch/ glibc/2.28
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False
```
</p></details>

<details open><summary><code>`conda config --show-sources`</code></summary><p>

```

```
</p></details>

<details><summary><code>`conda list --show-channel-urls`</code></summary><p>

```
# packages in environment at /scratch/bevan/anaconda3:
#
# Name                    Version                   Build  Channel
asn1crypto                0.24.0                   py37_0    defaults
ca-certificates           2018.03.07                    0    defaults
certifi                   2018.10.15               py37_0    defaults
cffi                      1.11.5           py37he75722e_1    defaults
chardet                   3.0.4                    py37_1    defaults
conda                     4.5.11                   py37_0    defaults
conda-env                 2.6.0                         1    defaults
cryptography              2.4.1            py37h1ba5d50_0    defaults
idna                      2.7                      py37_0    defaults
libedit                   3.1.20170329         h6b74fdf_2    defaults
libffi                    3.2.1                hd88cf55_4    defaults
libgcc-ng                 8.2.0                hdf63c60_1    defaults
libstdcxx-ng              8.2.0                hdf63c60_1    defaults
ncurses                   6.1                  hf484d3e_0    defaults
openssl                   1.1.1a               h7b6447c_0    defaults
pip                       18.1                     py37_0    defaults
pycosat                   0.6.3            py37h14c3975_0    defaults
pycparser                 2.19                     py37_0    defaults
pyopenssl                 18.0.0                   py37_0    defaults
pysocks                   1.6.8                    py37_0    defaults
python                    3.7.1                h0371630_3    defaults
readline                  7.0                  h7b6447c_5    defaults
requests                  2.20.1                   py37_0    defaults
ruamel_yaml               0.15.46          py37h14c3975_0    defaults
setuptools                40.6.2                   py37_0    defaults
six                       1.11.0                   py37_1    defaults
sqlite                    3.25.3               h7b6447c_0    defaults
tk                        8.6.8                hbc83047_0    defaults
urllib3                   1.23                     py37_0    defaults
wheel                     0.32.3                   py37_0    defaults
xz                        5.2.4                h14c3975_4    defaults
yaml                      0.1.7                had09818_2    defaults
zlib                      1.2.11               h7b6447c_3    defaults
```
</p></details>

## Additional comments

- I don't know if this is an issue with conda or with the anaconda package. However, I cannot point to an actual issue based on the output of conda. How can a single specification conflict? Does it conflict with itself?
- Debug output of `conda create --debug -n anaconda anaconda=5.3.1` is attached

[debug.log](https://github.com/conda/conda/files/2611539/debug.log)

**Comment by michaellass** [comment]
I can reproduce this issue on several linux installations. On macOS however I get a slightly different error message which may give additional information:

```
UnsatisfiableError: The following specifications were found to be in conflict:
  - anaconda=5.3.1 -> twisted==18.7.0=py27h1de35cc_1 -> pyobjc-core
  - anaconda=5.3.1 -> twisted==18.7.0=py27h1de35cc_1 -> pyobjc-framework-cfnetwork
  - anaconda=5.3.1 -> twisted==18.7.0=py27h1de35cc_1 -> pyobjc-framework-cocoa
```

However, installing twisted==18.7.0=py27h1de35cc_1 itself works without problems.

Again, the output with --debug is attached.

[debug.log](https://github.com/conda/conda/files/2611561/debug.log)

**Comment by nehaljwani** [comment]
The error, at least on Linux is due to:
```
Solving environment: failed

UnsatisfiableError: The following specifications were found to be in conflict:
  - mkl-service==1.1.2=py37h90e4bf4_5 -> mkl[version='>=2018.0.3,<2019.0a0']
  - mkl==2019.0=118
Use "conda info <package>" to see the dependencies for each package.
```

**Comment by nehaljwani** [comment]
This is a side effect of https://github.com/AnacondaRecipes/repodata-hotfixes/pull/7/files

cc @msarahan @soapy1

**Comment by michaellass** [comment]
> The error, at least on Linux is due to:
> 
> ```
> Solving environment: failed
> 
> UnsatisfiableError: The following specifications were found to be in conflict:
>   - mkl-service==1.1.2=py37h90e4bf4_5 -> mkl[version='>=2018.0.3,<2019.0a0']
>   - mkl==2019.0=118
> Use "conda info <package>" to see the dependencies for each package.
> ```

I'm wondering how you could get conda to output this information. However, the problem is solved for me both on Linux and on macOS, presumably by AnacondaRecipes/repodata-hotfixes#10. Thanks! 👍

**Comment by msarahan** [comment]
Yes, that was the solution.  We had been operating under the assumption that MKL was backwards compatible.  There were breaks in that compatibility in the 2019 release, though.  So, we had to make sure that there was an upper bound, but then that upper bound was incompatible with the MKL version we included in Anaconda 5.3.0.  MKL 2019 will be properly supported in the next anaconda release, and in our packages sooner.

**Comment by nehaljwani** [comment]
>I'm wondering how you could get conda to output this information. However, the problem is solved for me both on Linux and on macOS, presumably by AnacondaRecipes/repodata-hotfixes#10. Thanks! 👍

I ran this command:
```
conda create -p /path/does/not/exist $(conda search --info anaconda=5.3.1=py37_0 --json | jq -r '.anaconda[].depends[]' | sed -e 's/ /=/g' -e 's/\(.*\)/"\1"/g' | tr '\n' ' ')
```
